scripts/mysql_system_tables_fix.sql

```diff
@@ -1040,28 +1053,73 @@ ALTER TABLE help_keyword CONVERT TO CHARACTER SET utf8mb3;
 -- time_zone, time_zone_leap_second, time_zone_name, time_zone_transition,
 -- time_zone_transition_type, columns_priv, db, procs_priv, proxies_priv,
 -- tables_priv, user.
+SET @component_engine = (select ENGINE from information_schema.tables where
+    table_schema='mysql' and table_name='component');
+SET @cmd = CONCAT("ALTER TABLE component ENGINE=",
+    IF(@ddse = 'ROCKSDB', "ROCKSDB","InnoDB"));
+SET @str = IF(@ddse = @component_engine, "SET @dummy = 0", @cmd);
+PREPARE stmt FROM @str;
+EXECUTE stmt;
+DROP PREPARE stmt;
+
+SET @default_roles_engine = (select ENGINE from information_schema.tables where
+    table_schema='mysql' and table_name='default_roles');
+SET @cmd = CONCAT("ALTER TABLE default_roles ENGINE=",
+    IF(@ddse = 'ROCKSDB', "ROCKSDB","InnoDB"));
+SET @str = IF(@ddse = @default_roles_engine, "SET @dummy = 0", @cmd);
+PREPARE stmt FROM @str;
+EXECUTE stmt;
+DROP PREPARE stmt;
+
+SET @engine_cost_engine = (select ENGINE from information_schema.tables where
+    table_schema='mysql' and table_name='engine_cost');
+SET @cmd = CONCAT("ALTER TABLE engine_cost ENGINE=",
+    IF(@ddse = 'ROCKSDB', "ROCKSDB","InnoDB"));
+SET @str = IF(@ddse = @engine_cost_engine, "SET @dummy = 0", @cmd);
+PREPARE stmt FROM @str;
+EXECUTE stmt;
+DROP PREPARE stmt;
+
```

Comment:
This and the other chunks below mix the engine change with the existing `ALTER
TABLE ENGINE=...` statements for other tables. I don't think this is right,
because the purpose and the engine choice logic is different. The existing
script looks at the DDSE sysvar, and the added logic looks at the actual engine.

I believe the existing section should be left as-is, and a new section should be
introduced, not sure if above or below this one, and it should also have `ALTER
TABLE` for the tables mentioned in this existing section.

Reply:
Move all of these into its own section.